src/battle_util.c

```diff
@@ -1885,6 +1901,8 @@ u8 CheckMoveLimitations(u8 battler, u8 unusableMoves, u16 check)
         // Gorilla Tactics
         else if (check & MOVE_LIMITATION_CHOICE_ITEM && GetBattlerAbility(battler) == ABILITY_GORILLA_TACTICS && *choicedMove != MOVE_NONE && *choicedMove != MOVE_UNAVAILABLE && *choicedMove != gBattleMons[battler].moves[i])
             unusableMoves |= gBitTable[i];
+        else if (check & MOVE_LIMITATION_GIGATON_HAMMER && gBattleMoves[gBattleMons[battler].moves[i]].effect == EFFECT_GIGATON_HAMMER && gDisableStructs[battler].gigatonHammerTimer)
```

Comment:
It functions the same as Torment, so there's really no need for an extra timer.

Reply:
Would this work? Torment applies to a battler so no move can be used twice in a row.

Reply:
> It functions the same as Torment, so there's really no need for an extra timer.

It kinda doesn't though. Torment affects a battler and forbids them from using X or Y move twice in a row until they switch out, and the way that's accomplished is by branding the battler in question with a full status flag, `STATUS2_TORMENT`.

Gigaton Hammer has a cooldown period essentially; you can't use that specific move twice in a row, that's why the timer works just fine instead of wasting a full status flag on it for no good reason :eyes: ...

Reply:
Alex understood what I meant. With a timer its not implemented correctly. See [this post](https://www.smogon.com/forums/threads/scarlet-violet-battle-mechanics-research.3709545/post-9412011) explaining why. With the timer implementation, a Pokémon would be allowed to select it again after being asleep which is not correct.

Reply:
Removed the Gigaton Hammer timer.

Reply:
Looks good now, just do a test to see if it should be gLastMoves or gLastResultingMoves. gLastMoves may not let it be selected after being upgraded to a Z-Move.

Reply:
You also may want to implement the new Blood Moon attack while you have this open.

Reply:
> gLastResultingMoves

Yep, good call, `gLastResultingMoves` should be used.

Reply:
> You also may want to implement the new Blood Moon attack while you have this open.

2-week spoiler policy 👀

Reply:
> > You also may want to implement the new ___________ attack while you have this open.
> 
> 2-week spoiler policy 👀

In this case I would say that ghoul should decide whether to wait those 2 weeks for the move in question or merge before it.

Reply:
> In this case I would say that ghoul should decide whether to wait those 2 weeks for the move in question or merge before it.

If we set Tackle with `EFFECT_GIGATON_HAMMER` and then using one after the other works, the effect itself should be ready, and in 2 weeks we only need to add the new move data.

Reply:
Yeah, already checked. Works fine with 2 moves of the same effect on a moveset.